### PR TITLE
SC2: Don't Filter Excluded Victory Locations

### PR DESCRIPTION
**Pull request by Alchav**
## What is this fixing or adding?
In testing #3017 I put a victory location into `exclude_locations` to make sure it worked. I found the location was not created. Evidently SC2 filters out excluded locations, and attempts to prevent victory locations from being filtered out, but fails due to a bug. I didn't check what happens if a game is played under this condition but since victory locations determine whether a mission is considered beaten, I can't imagine it goes well.

## How was this tested?
Generating with an excluded victory location and observing in the spoiler log that the location is still created.

### CODE REVIEW

**Comment by Alchav** [comment]
@Ziktofel

**Review by Ziktofel — APPROVED** [review]

**Review by neocerber — APPROVED** [review]
As author stated, the intent of the code modified was to always include Locations of the Victory type in a SC2 world. The old code failed to achieve this since it was checking if the class/struct LocationType was a Victory (which it is not XD). The proposed changes does what is intended: check if the type of a given Location is of the Victory type. Even if it is straightforward, I also repeated the author test and confirms that it does what it is intended to do. Furthermore, without this change, the generation failed in my tests when the excluded Location was a mission that is required to access a good part of the mission table.

tl;dr correct the bug correctly.

**Review by Berserker66 — APPROVED** [review]
